src/view_maybe/view_maybe.h

```diff
@@ -8,130 +8,91 @@
-
-
-template<class T>
-concept bool ContextualBool =
-    requires(const T& t) {
-    {bool(t)} -> bool;
-};
+template <class T, class Ref, class ConstRef>
+concept bool _Nullable2 =
+    std::is_lvalue_reference_v<Ref> &&
+    std::is_object_v<std::remove_reference_t<Ref>> &&
+    std::is_lvalue_reference_v<ConstRef> &&
+    std::is_object_v<std::remove_reference_t<ConstRef>> &&
+    ranges::ConvertibleTo<std::add_pointer_t<ConstRef>, 
+        const std::remove_reference_t<Ref>*>;
 
 template <class T>
 concept bool Nullable =
-    std::experimental::ranges::Readable<std::remove_reference_t<T>> &&
-    ContextualBool<T> &&
-    requires (const T& t) {
-    typename dereference_t<T>;
-    std::is_object_v<dereference_t<T>>;
-};
-
+    std::is_object_v<T> &&
+    requires(T& t, const T& ct) {
+        bool(ct);
+        *t;
+        *ct;
+    } &&
+    _Nullable2<T, ranges::iter_reference_t<T>, ranges::iter_reference_t<const T>>;
 
 template <Nullable Maybe>
 class safe_maybe_view
     : public std::experimental::ranges::view_interface<safe_maybe_view<Maybe>> {
   private:
-    using T = std::decay_t<std::remove_reference_t<dereference_t<Maybe>>>;
-    using M = std::remove_cv_t<std::remove_reference_t<Maybe>>;
+    using T = std::remove_reference_t<ranges::iter_reference_t<Maybe>>;
 
-    std::experimental::ranges::detail::semiregular_box<M> value_;
+    std::experimental::ranges::detail::semiregular_box<Maybe> value_;
 
   public:
-    safe_maybe_view() = default;
+    constexpr safe_maybe_view() = default;
+    constexpr explicit safe_maybe_view(Maybe const& maybe) 
+        noexcept(std::is_nothrow_copy_constructible_v<Maybe>)
+        : value_(maybe) {}
+    constexpr explicit safe_maybe_view(Maybe&& maybe) 
+        noexcept(std::is_nothrow_move_constructible_v<Maybe>)
+        : value_(std::move(maybe)) {}
 
-    constexpr explicit safe_maybe_view(Maybe const& maybe) : value_(maybe) {}
-    constexpr explicit safe_maybe_view(Maybe&& maybe) : value_(std::move(maybe)) {}
     constexpr T*       begin() noexcept { return data(); }
     constexpr const T* begin() const noexcept { return data(); }
-    constexpr T*       end() noexcept {
-        if (data())
-            return data() + 1;
-        else
-            return data();
-    }
+    constexpr T*       end() noexcept { return data() + size(); }
+    constexpr const T* end() const noexcept { return data() + size(); }
 
-    constexpr const T* end() const noexcept {
-        if (data())
-            return data() + 1;
-        else
-            return data();
-    }
-
-    constexpr std::ptrdiff_t size() noexcept {
-        if (value_.get())
-            return 1;
-        else
-            return 0;
-    }
+    constexpr std::ptrdiff_t size() noexcept { return bool(value_.get()); }
 
     constexpr T* data() noexcept {
-        if (value_.get())
-            return std::addressof(*value_.get());
+        Maybe& m = value_.get();
+        if (m)
```

Comment:
I didn't change these since I figured it was a style choice, but I'll point out they could be:
```c++
return m ? std::addressof(*m) : nullptr;
```
(The conditional operator does contextually convert its first argument to `bool`.)